pumps/sql.go

```diff
@@ -201,3 +216,92 @@ func (c *SQLPump) WriteData(ctx context.Context, data []interface{}) error {
 
 	return nil
 }
+
+func (c *SQLPump) WriteUptimeData(data []interface{}) {
+	c.log.Debug("Attempting to write ", len(data), " records...")
+
+	var typedData []analytics.UptimeReportData
+
+	batch := 500
+	newBatch := false
+	var firstDate time.Time
+	for i, r := range data {
+
+		decoded := analytics.UptimeReportData{}
+
+		if err := msgpack.Unmarshal([]byte(r.(string)), &decoded); err != nil {
+			c.log.Error("Couldn't unmarshal uptime analytics data:", err)
+
+			continue
+		}
+		if i == 0 {
+			firstDate = decoded.TimeStamp
+		}
+
+		typedData = append(typedData, decoded)
+
+		if c.SQLConf.TableSharding && !newBatch && firstDate.Format("20060102") < decoded.TimeStamp.Format("20060102") {
+			batch = i
+			newBatch = true
+		}
+	}
+
+	if c.SQLConf.TableSharding && len(typedData) > 0 {
```

Comment:
As we discussed in pair programming, we can get rid of this if statement.

Reply:
solved!